PyHT6022/HantekFirmware/custom/device.c

```diff
@@ -158,30 +171,32 @@ void select_interface(BYTE alt)
 	EP2AUTOINLENL = pPacketSize[0];
 	EP2AUTOINLENH = pPacketSize[1] & 0x7;
 	EP2ISOINPKTS = (pPacketSize[1] >> 3) + 1;
+	set_aadj();
     }
 }
 
 const struct samplerate_info {
     BYTE rate;
+    BYTE efrate;
     BYTE wait0;
     BYTE wait1;
     BYTE opc0;
     BYTE opc1;
     BYTE out0;
     BYTE ifcfg;
 } samplerates[] = {
-    { 48,0x80,   0, 3, 0, 0x00, 0xea },
-    { 30,0x80,   0, 3, 0, 0x00, 0xaa },
-    { 24,   1,   0, 2, 1, 0x40, 0xca },
-    { 16,   1,   1, 2, 0, 0x40, 0xca },
-    { 12,   2,   1, 2, 0, 0x40, 0xca },
-    {  8,   3,   2, 2, 0, 0x40, 0xca },
-    {  4,   6,   5, 2, 0, 0x40, 0xca },
-    {  2,  12,  11, 2, 0, 0x40, 0xca },
-    {  1,  24,  23, 2, 0, 0x40, 0xca },
-    { 50,  48,  47, 2, 0, 0x40, 0xca },
-    { 20, 120, 119, 2, 0, 0x40, 0xca },
-    { 10, 240, 239, 2, 0, 0x40, 0xca }
+    { 48, 48, 0x80,   0, 3, 0, 0x00, 0xea },
+    { 30, 48, 0x80,   0, 3, 0, 0x00, 0xaa },
+    { 24, 24,    1,   0, 2, 1, 0x40, 0xca },
+    { 16, 16,    1,   1, 2, 0, 0x40, 0xca },
+    { 12, 12,    2,   1, 2, 0, 0x40, 0xca },
+    {  8,  8,    3,   2, 2, 0, 0x40, 0xca },
+    {  4,  4,    6,   5, 2, 0, 0x40, 0xca },
+    {  2,  2,   12,  11, 2, 0, 0x40, 0xca },
+    {  1,  1,   24,  23, 2, 0, 0x40, 0xca },
+    { 50,  1,   48,  47, 2, 0, 0x40, 0xca },
```

Comment:
Maybe it is better to make efrate a WORD and store rate/100KHz, i.e. 480,240,...,5,2,1.